Prox/Prox/FirebasePlacesDatabase.swift

```diff
@@ -88,12 +84,15 @@ class FirebasePlacesDatabase: PlacesDatabase {
 
         let childRef = placeDetailsRef.child(placeKey)
         childRef.queryOrderedByKey().observeSingleEvent(of: .value) { (data: FIRDataSnapshot) in
+            guard data.exists() else {
```

Comment:
We could probably make this a little more concise like this
```
            guard data.exists(),
                let place = Place(fromFirebaseSnapshot: data) else {
                    let message = data.exists() ? "Child place key does not exist: \(placeKey)" : "Snapshot missing required Place data: \(data)"
                    deferred.fill(with: DatabaseResult.fail(withMessage: message))
                    return

            }
            deferred.fill(with: DatabaseResult.succeed(value: place))
```
But seeing as it works just fine this way, I see no reason to change it.